packages/wdio-cucumber-framework/src/utils.ts

```diff
@@ -198,6 +198,12 @@ export function addKeywordToStep(steps:ReporterStep[], feature:messages.GherkinD
             // the second from the TableRow AST node.
             // See https://github.com/cucumber/cucumber/blob/master/messages/messages.md
             const astNodeId = step.astNodeIds[0]
+
+            const rules  = feature.children.filter((child)=> Object.keys(child)[0]=== 'rule')
+            feature.children = feature.children.filter((child)=> Object.keys(child)[0]!== 'rule')
+            const rulesChildrens:any = rules.map((child)=>child.rule?.children).flat()
```

Comment:
```suggestion
            const rulesChildrens = rules.map((child) => child.rule?.children).flat()
```

Reply:
maybe it makes sense to change the line to `rules.filter((child) => child.rule).map((child) => child.rule?.children).flat()`?
Otherwise, it's possible to have an array of `undefined` values in case there is no `child.rule`

Reply:
@mgrybyk  rules will have rules , the structure is like { rules : rule :children}, if there is no rules rules will be an empty array

Reply:
Let's don't use `. ? ` in such case

Reply:
@mgrybyk true ,  i will check the type for it . IT was giving type script error thank you :)